design/decouple-ceph-version.md

```diff
@@ -0,0 +1,88 @@
+
+### Rook Version
+
+The Rook version is defined by the operator's container `image` tag. All Rook containers launched by the operator will also launch the same version of the Rook image.
+The full image name is an important part of the version. This allows the container to be loaded from a private repo if desired.
+
+In this example, the Rook version is `rook/ceph:v0.8.1`.
+
+```yaml
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: rook-ceph-operator
+spec:
+  template:
+    spec:
+      containers:
+      - name: rook-ceph-operator
+        image: rook/ceph:v0.8.1
+...
+```
+
+### Ceph Version
+
+The Ceph version is defined as the property `cephImage` in the Cluster CRD. All Ceph daemon containers launched by the Rook operator will use this image, including the mon, mgr,
+osd, rgw, and mds pods. The significance of this approach is that the Rook binary is not included in the daemon containers. All initialization performed by Rook to initialize and generate the Ceph config must be completed in an [init container](https://github.com/rook/rook/issues/2003). Once the Rook init containers complete their execution, the daemon container can run the Ceph image. The daemon container will no longer have Rook running.
+
+In the following Cluster CRD example, the Ceph version is Mimic 13.2.1.
+
+```yaml
+apiVersion: ceph.rook.io/v1beta1
+kind: Cluster
+metadata:
+  name: rook-ceph
+  namespace: rook-ceph
+spec:
+  cephImage: ceph/ceph:v13.2.1
+...
+```
+
+### Operator Requirements
+
+The operator needs to run the Ceph client tools to manage the cluster. For example, the `ceph` tool is needed for general Ceph configuration and status, `radosgw-admin` is required for managing an object store. Therefore, all the necessary client tools will still be included in the Rook image.
+
+The client tools are tested by the Ceph team to be backward and forward compatible by two versions. Therefore, the operator can support a version of Ceph up to two versions older than the client tools it contains. The operator will also be able to orchestrate upgrading to two versions newer of Ceph, in case there is a need to connect the operator to a newer version of a daemon before the operator is updated.
+
+The operator will be made to understand differences in the Ceph versions that are necessary for orchestration. Some examples might include:
+- If running Luminous, start the Ceph dashboard on http. If running Mimic, a self-signed cert could be generated to start the dashboard with https.
+- If a new daemon is added in a future Ceph release, the operator would understand to deploy that daemon only if the Ceph version is at least that version.
+
+### Upgrades
+
+The flexibility during upgrades will now be improved since the upgrade of Rook will be independent from the upgrade to the Ceph version.
+- To upgrade Rook, update the version of the Rook operator container
+- To upgrade Ceph, update the `cephImage` in the cluster CRD
```

Comment:
Maybe add a "then," in here to indicate the general procedure is to first upgrade the operator, then ceph?